pupil_src/shared_modules/marker_tracker_3d/marker_detector.py

```diff
@@ -1,27 +1,29 @@
 import logging
 
 import numpy as np
+
 import square_marker_detect
 
 logger = logging.getLogger(__name__)
 
 
 class MarkerDetector:
-    def __init__(self, storage, min_marker_perimeter):
+    def __init__(self, storage):
         self.storage = storage
-        self.min_marker_perimeter = min_marker_perimeter
 
     def detect(self, frame):
         # not use detect_markers_robust to avoid cv2.calcOpticalFlowPyrLK for
         # performance reasons
-        markers = square_marker_detect.detect_markers(
-            frame.gray,
-            grid_size=5,
-            aperture=13,
-            min_marker_perimeter=self.min_marker_perimeter,
-        )
-        markers_dict = self._filter_markers(markers)
-        self.storage.markers = markers_dict
+        try:
+            markers = square_marker_detect.detect_markers(
+                frame.gray,
+                grid_size=5,
+                aperture=13,
+                min_marker_perimeter=self.storage.min_marker_perimeter,
+            )
+            self.storage.markers = self._filter_markers(markers)
+        except AttributeError:
```

Comment:
What could cause an `AttributeError` here? Usually it is best to make the part of code covered in the `try`-part as small as possible, ideally just a single command. Otherwise chances increase that we catch an Error that was raised somewhere we did not expect it to be raised. We than catch it anyway and thus hide away bugs. So if possible shrink the try-part to only one of the commands. Note that you can also use try-except-else, where else will be executed after try only if no error was raised.

Reply:
You are right!